### ISSUE TITLE
[RFE] Support regexp or "selection expression" in the MatchLabels.

### ISSUE THREAD

**Issue by jtudelag** [redhat-cop/namespace-configuration-operator#21]
First of all, thanks for this awesome operator, it is super useful!!!

It would be great to be able to use regexp or "selection expressions" in the matchLabel selector

For example:
```
selector:
    matchLabels:
      regexp:
         ".*-dev"
```

Or selection expressions, like the ones used in the [OCP Router Sharding](https://docs.openshift.com/container-platform/3.11/install_config/router/default_haproxy_router.html#using-router-shards):

```
selector:
    matchLabels:
      operand:
        "name in (dev, ops)"
```

**Comment by stocky37** [comment]
You can use `matchExpressions` instead of `matchLabels` for selection expressions. See: [Resources that support set based requirements](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#resources-that-support-set-based-requirements).

**Comment by raffaelespazzoli** [comment]
closed by #22
